clusterous/cluster.py

```diff
@@ -1107,7 +1176,7 @@ def _get_controller_private_ip(self):
                 break
         return ip
 
-    def add_nodes(self, num_nodes, instance_type, node_name):
+    def add_nodes(self, num_nodes, instance_type, node_name, spot_price=0):
```

Comment:
Given that add_nodes and rm_nodes shouldn't be specific to cloud provider, maybe there shouldn't be an AWS specific param. Perhaps have a generic "params" dictionary that can contain cloud specific info such as whether this is a spot request, etc.